client/arg.cpp

```diff
@@ -159,7 +159,18 @@ bool analyse_argv(const char * const *argv, CompileJob &job, bool icerun, list<s
                 /* as above but with extra argument */
             } else if (!strcmp(a, "-MT") || !strcmp(a, "-MQ")) {
                 args.append(a, Arg_Local);
-                args.append(argv[++i], Arg_Local);
+                ++i;
+                if (argv[i][0] == '"') {
```

Comment:
I agree, I wanted to minimize changes but I think it would be best to check all arguments for quotes, just in case. I think it would be best to have a function that returns the argument value that checks for quotes. Note that we call icecream by wrapping it with a shell script that sets the appropriate environment variables, so it's possible that by escaping them we could get right values passed to icecc. I'll check and revisit this patch.